packages/core/integration-tests/test/transpilation.js

```diff
@@ -0,0 +1,215 @@
+
+describe('transpilation', function() {
+  it('should not transpile if no targets are defined', async function() {
+    await bundle(path.join(__dirname, '/integration/babel-default/index.js'), {
+      defaultTargetOptions: {
+        engines: undefined,
+        shouldOptimize: false,
+      },
+    });
+    let file = await outputFS.readFile(path.join(distDir, 'index.js'), 'utf8');
+    assert(file.includes('class Foo'));
+    assert(file.includes('class Bar'));
+  });
+
+  it('should support transpiling using browserlist', async function() {
+    await bundle(
+      path.join(__dirname, '/integration/babel-browserslist/index.js'),
+    );
+
+    let file = await outputFS.readFile(path.join(distDir, 'index.js'), 'utf8');
+    assert(file.includes('function Foo'));
+    assert(file.includes('function Bar'));
+  });
+
+  it('should support transpiling when engines have semver ranges', async () => {
+    let fixtureDir = path.join(__dirname, '/integration/babel-semver-engine');
+    await bundle(path.join(fixtureDir, 'index.js'));
+
+    let legacy = await outputFS.readFile(
+      path.join(fixtureDir, 'dist', 'legacy.js'),
+      'utf8',
+    );
+    assert(legacy.includes('function Foo'));
+    assert(legacy.includes('function Bar'));
+
+    let modern = await outputFS.readFile(
+      path.join(fixtureDir, 'dist', 'modern.js'),
+      'utf8',
+    );
+    assert(modern.includes('class Foo'));
+    assert(modern.includes('class Bar'));
+  });
+
+  it('should not transpile node_modules by default', async function() {
+    await bundle(
+      path.join(__dirname, '/integration/babel-node-modules/index.js'),
+    );
+
+    let file = await outputFS.readFile(path.join(distDir, 'index.js'), 'utf8');
+    assert(/class \S+ \{/.test(file));
+    assert(file.includes('function Bar'));
+  });
+
+  it('should not compile node_modules with a source field in package.json when not symlinked', async function() {
+    await bundle(
+      path.join(
+        __dirname,
+        '/integration/babel-node-modules-source-unlinked/index.js',
+      ),
+    );
+
+    let file = await outputFS.readFile(path.join(distDir, 'index.js'), 'utf8');
+    assert(!file.includes('function Foo'));
+    assert(file.includes('function Bar'));
+  });
+
+  it('should support compiling JSX', async function() {
+    await bundle(path.join(__dirname, '/integration/jsx/index.jsx'));
+
+    let file = await outputFS.readFile(path.join(distDir, 'index.js'), 'utf8');
+    assert(file.includes('React.createElement("div"'));
+  });
+
+  it('should support compiling JSX in JS files with React dependency', async function() {
+    await bundle(path.join(__dirname, '/integration/jsx-react/index.js'));
+
+    let file = await outputFS.readFile(path.join(distDir, 'index.js'), 'utf8');
+    assert(file.includes('React.createElement("div"'));
+  });
+
+  it.skip('should support compiling JSX with pure annotations', async function() {
+    // TODO: SWC
```

Comment:
(I have a branch that instead uses the swc react preset, just waiting for https://github.com/swc-project/swc/pull/1639 to be released.)